client.go

```diff
@@ -111,10 +111,9 @@ func (c *Client) request(method, requestPath string, query url.Values, body []by
 			continue
 		}
 
-		defer resp.Body.Close()
-
 		// read the body (even on non-successful HTTP status codes), as that's what the unit tests expect
 		bodyContents, err = ioutil.ReadAll(resp.Body)
+		_ = resp.Body.Close()
```

Comment:
I think I prefer this instead, so it's easier to `grep` in the future:
```suggestion
		resp.Body.Close() //nolint:errcheck
```